src/metabase/lib/convert.cljc

```diff
@@ -553,8 +553,18 @@
                                                            ;; direction. Also, `:aggregation-options` is not allowed
                                                            ;; inside `:expressions` in legacy, we'll just have to toss
                                                            ;; the extra info.
-                                                           (if (#{:value :aggregation-options} (first legacy-clause))
+                                                           (cond
+                                                             (= (first legacy-clause) :aggregation-options)
                                                              (second legacy-clause)
+
+                                                             ;; do not unwrap `nil` `:value` clauses, these are used
+                                                             ;; internally by the Pivot QP and unwrapping them breaks
+                                                             ;; stuff
+                                                             (and (= (first legacy-clause) :value)
+                                                                  (some? (second legacy-clause)))
+                                                             (second legacy-clause)
```

Comment:
Not super sure we should be unwrapping `:value` clauses at all, I'm pretty sure `:value` is almost exclusively used internally by the QP and if we're converting back and forth between legacy and pMBQL we should probably preserve any places we have a `:value` clause during the conversion. But I didn't want to second guess the code here so I made my exception as small as possible for now